android/anc/src/main/java/org/smartregister/fhircore/anc/ui/reports/ReportsHomeScreen.kt

```diff
@@ -0,0 +1,204 @@
+import androidx.compose.material.IconButton
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TopAppBar
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.paging.LoadState
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.PagingSource
+import androidx.paging.PagingState
+import androidx.paging.compose.collectAsLazyPagingItems
+import androidx.paging.compose.itemsIndexed
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import org.smartregister.fhircore.anc.R
+import org.smartregister.fhircore.anc.data.report.model.ReportItem
+import org.smartregister.fhircore.engine.ui.theme.AppTheme
+import org.smartregister.fhircore.engine.ui.theme.SubtitleTextColor
+import org.smartregister.fhircore.engine.util.annotation.ExcludeFromJacocoGeneratedReport
+
+@Composable
+fun ReportsHomeScreen(dataProvider: ReportDataProvider) {
+  Surface(color = colorResource(id = R.color.white)) {
+    Column {
+
+      // top bar
+      TopAppBar(
+        title = { Text(text = stringResource(id = R.string.reports)) },
+        navigationIcon = {
+          IconButton(onClick = { dataProvider.getAppBackClickListener().invoke() }) {
+            Icon(Icons.Filled.ArrowBack, contentDescription = "Back arrow")
+          }
+        }
+      )
+
+      val lazyEncounterItems = dataProvider.getReportsTypeList().collectAsLazyPagingItems()
+
+      LazyColumn(modifier = Modifier.background(Color.White).fillMaxSize()) {
+        itemsIndexed(lazyEncounterItems) { _, item -> ReportRow(item!!, { _, _ -> }) }
+
+        lazyEncounterItems.apply {
+          when {
+            loadState.refresh is LoadState.Loading -> {
+              item { LoadingItem() }
+            }
+            loadState.append is LoadState.Loading -> {
+              item { LoadingItem() }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+@Composable
+fun LoadingItem() {
+  CircularProgressIndicator(
+    modifier =
+      Modifier.testTag("ProgressBarItem")
+        .fillMaxWidth()
+        .padding(16.dp)
+        .wrapContentWidth(Alignment.CenterHorizontally)
+  )
+}
+
+@Preview
+@Composable
+@ExcludeFromJacocoGeneratedReport
+fun ReportsHomeScreenPreview() {
+  AppTheme { ReportsHomeScreen(dummyData()) }
+}
+
+fun dummyData() =
```

Comment:
This function is not covered in any nest. It's used in the composable function `ReportsHomeScreenPreview` that is `ExcludeFromJacocoGeneratedReport`. Add tests for `ReportsHomeScreen` to cover the function as well.